Utils/azuretools-core/src/com/microsoft/azuretools/adauth/AuthContext.java

```diff
@@ -128,6 +135,73 @@ public AuthResult acquireToken(@NotNull String resource, boolean newAuthCode,
         }
     }
 
+    /**
+     * Acquire token for resource by device code.
+     */
+    public AuthResult acquireToken(@NotNull final String resource, final boolean newDeviceCode, final String uid,
+                                   final AuthenticationCallback<AuthenticationResult> callback) throws AuthException {
+        try {
+            if (!newDeviceCode) {
+                final AuthResult result = acquireTokenFromCache(resource, uid);
+                if (result != null) {
+                    return result;
+                } else {
+                    throw new AuthException(AuthError.UnknownUser);
+                }
+            }
+            final ExecutorService service = Executors.newFixedThreadPool(5);
```

Comment:
Looks like we need to shutdown it by hand, otherwise, we'll get lots of them alive.

Reply:
Yes, I find the recommended way to shut down the service. Will update.